components/PreparePulls.coffee

```diff
@@ -1,4 +1,24 @@
 noflo = require 'noflo'
+uuid = require 'uuid'
+
+sendPulls = (pulls, repo, scope, output, callback) ->
+  unless pulls.length
+    do callback
+    return
+  entry = pulls.shift()
```

Comment:
Still has mixed app logic and throttling logic

Reply:
Yep, let me split to a separate component

Reply:
@bergie just use bluebird.mapSeries or bluebird.map with concurrency param?

Reply:
That would involve adding a new dependency

Reply:
On another thought, let me keep it as-is for now. We should figure out a proper way to do throttling with Process API on NoFlo level

NoFlo knows the "load" of the component, but unlike legacy AsyncComponent, we're not exposing this to the network in any way

Reply:
If it is not important to show partial results, I don't think there are any benefits from doing throttling on NoFlo level

Reply:
Some notes: https://github.com/noflo/noflo/issues/581